src/parser.ts

```diff
@@ -669,13 +669,13 @@ export class Parser {
                     } else if (this.matchKeyword('class')) {
                         expr = this.parseClassExpression();
                     } else {
-                        this.throwUnexpectedToken(this.nextToken());
+                        throw this.unexpectedTokenError(this.nextToken());
```

Comment:
Whoa, thanks for that `never` trick!
I'll revert those inlined throws and just use your suggested approach.